pkg/fs/dir.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019 The virnsnap authors. See file "AUTHORS".
+// Licensed under the MIT License. You have obtained a copy of the License at
+// the "LICENSE" file in this repository.
+
+// Package fs implements helper functions for handling filesystem related
+// tasks.
+package fs
+
+import (
+	"fmt"
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+// EnsureDirectory is a helper function that ensures path points to a directory.
+// If the directory does not exist, create it. The directory needs to be
+// accessible and writeable so the check can pass.
+func EnsureDirectory(path string) error {
+	stat, err := os.Stat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = os.Mkdir(path, 0700)
```

Comment:
I think it makes sense to define the default directory mode as a constant at the beginning of the package.